src/main/java/com/microsoft/aad/msal4j/IHttpClient.java

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+/**
+ * Interface to be implemented when configuring http client for {@link IPublicClientApplication} or
+ * {@link IConfidentialClientApplication}.
+ */
+public interface IHttpClient {
```

Comment:
>IHttpClient [](start = 17, length = 11)

Seems like a lot of the methods in the DefaultHttpClient implementation is not available in this interface, is that on purpose?

Reply:
Yea, those are implementation details . For customers configuring their own IHttpClient, all we care about is that they `send` the `HttpRequest` that we pass to them and they return an `IHttpResponse` - everything else is implementation details we are not concerned with. 

If no HTTP client is configured, we use JDK `HttpsUrlConnection` for making HTTP calls (in `DefaultHttpClient`). Everything other than `send` are implementation details specific to JDKs `HttpsUrlConnection`